spec.md

```diff
@@ -2027,19 +2049,16 @@ Since they are intended for verification, ports with a probe type do not necessa
 ## Types
 
 There are two probe types, `Probe<T>`{.firrtl} is a read-only variant and `RWProbe<T>`{.firrtl} is a read-write variant.
-(See [@sec:probe-types]).
+Probes may be layer-colored.
+See [@sec:probe-types] for more information.
 
-Examples:
+## Layer Coloring
 
-``` firrtl
-Probe<UInt<8>>
-RWProbe<UInt<8>>
-```
-
-A `RWProbe<T>`{.firrtl} will be implicitly cast to a `Probe<T>`{.firrtl} whenever an expression of type `RWProbe<T>`{.firrtl} is assigned to a port with type `Probe<T>`{.firrtl}.
+Probe types may be colored with a layer (see [@sec:layers]).
+A layer-colored probe type places restrictions on the operations which use it.
 
-While `RWProbe<T>`{.firrtl} is strictly more powerful, `Probe<T>`{.firrtl} allows for more aggressive optimization.
-You should consider using `Probe`{.firrtl} whenever you know you don't need to ability to force (see [@sec:forcing]).
+An operation may only "read" from a probe whose layer color is enabled when the operation is enabled.
+An operation that "writes" to a probe must be in a region whose layer color is enabled when the probe's layer color is enabled.
```

Comment:
These are the two most important sentences in this PR.

Give these a lot of scrutiny @mmaloney-sf.

Reply:
Is there a meaningful difference between a layer and a layer color?

My impression is the term "color" is just a colorful way to speak of the association between a probe and the layer it works against.

Both of these sentences also confuse me a bit, trying to interpret them. Trying to dig into why:

* What does it mean for "the operation [to be] enabled"?
* The quotes around "read" and "write" make me wonder if these are formal statements or if they should be taken intuitively.
* You use the word "region" -- Should this be "layer" instead? Eg, "must be in a layer whose color is enabled".
* The phrase "whose color is enabled" is also weird. Intuitively, colors can be mixed, but not enabled or disabled.
* I feel like there's some subtle compound condition in the "write" restriction.